components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -305,21 +302,14 @@ void PageContentExtractor::OnJSTranscriptUrlResult(
 
 void PageContentExtractor::GetSearchSummarizerKey(
     mojom::PageContentExtractor::GetSearchSummarizerKeyCallback callback) {
-  auto snapshotter = render_frame()->CreateAXTreeSnapshotter(
-      ui::AXMode::kWebContents | ui::AXMode::kHTMLMetadata);
-  ui::AXTreeUpdate snapshot;
-  snapshotter->Snapshot(0, {}, &snapshot);
-
-  RE2 pattern(kSearchSummarizerKeyPattern);
-  for (const auto& meta : snapshot.tree_data.metadata) {
-    std::string key;
-    if (re2::RE2::FullMatch(meta, pattern, &key)) {
-      std::move(callback).Run(key);
-      return;
-    }
+  auto element =
+      render_frame()->GetWebFrame()->GetDocument().Head().QuerySelector(
+          "meta[name=summarizer-key]");
+  if (element.IsNull()) {
+    std::move(callback).Run({});
+    return;
   }
-
-  std::move(callback).Run({});
+  std::move(callback).Run(element.GetAttribute("content").Utf8());
```

Comment:
Is this ok if `element.GetAttribute("content")` doesn't exist and returns `g_null_atom` (whatever that is!)? Do we need to check `HasAttribute` first?

Reply:
I don't think any of the test cases have a test element without the content attribute

Reply:
Thought I had one but turns out I missed adding it somehow, it appears to be empty string as expected btw.